src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetMappableFilterColumnOptionsModal.tsx

```diff
@@ -3,45 +3,53 @@ import Modal from '@common/components/Modal';
 import SummaryList from '@common/components/SummaryList';
 import SummaryListItem from '@common/components/SummaryListItem';
 import Pagination from '@common/components/Pagination';
+import VisuallyHidden from '@common/components/VisuallyHidden';
 import React, { useMemo, useState } from 'react';
 import chunk from 'lodash/chunk';
 
-const itemsPerPage = 10;
-
 interface Props {
-  id: string;
+  column: string;
   label: string;
-  modalLabel: string;
   options: string[];
+  pageSize?: number;
+  publicId?: string;
```

Comment:
Is there a chance of publicId being null?  Assuming that the column, label, publicId etc are all from a *source* Filter, we should always have a publicId in the mapping data.

Reply:
This modal is displayed for both filter columns that have been added or deleted. Where a filter column has been added, it doesn't actually have a public ID yet so it will be undefined as we won't have set it - see `ApiDataSetNewFilterColumnsTable`.